// pages/auth/Auth.js
import React, {useState} from 'react'
import {Link} from 'react-router-dom'

import useFetch from 'hooks/useFetch'

function Auth() {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [{isLoading, response, error}, doFetch] = useFetch('/users/login')

  /**
   * We want to fetch data from different URLs. This means that we want to define the URL
   * We want to make fetch sometimes on initialize (like in global feed) and sometimes on form submit (like in login form)
   * We want to get the isLoading state to know that we are in the process of getting data and show it to the user accordingly
   * We want to get errors to show something to the user if the API call is not successful
   */

  const onSubmit = (event) => {
    event.preventDefault()
    doFetch({
      method: 'post',
      data: {
        user: {
          email: 'r@ro.io',
          password: '12345678',
        },
      },
    })
  }

  return (
    <div className="auth-page">
      <div className="container page">
        <div className="row">
          <div className="col-md-6 offset-md-3 col-xs-12">
            <div className="text-xs-center">
              <h1>Login</h1>
              <p>
                <Link to="/register">Dont have an account?</Link>
              </p>
              <form onSubmit={onSubmit}>
                <fieldset>
                  <fieldset className="form-group">
                    <input
                      type="email"
                      className="form-control form-control-lg"
                      placeholder="Email"
                      value={email}
                      onChange={(event) => setEmail(event.target.value)}
                    />
                  </fieldset>
                  <fieldset className="form-group">
                    <input
                      type="password"
                      className="form-control form-control-lg"
                      placeholder="Password"
                      value={password}
                      onChange={(event) => setPassword(event.target.value)}
                    />
                  </fieldset>
                  <button
                    disabled={isLoading}
                    className="btn btn-lg btn-primary"
                    type="submit"
                  >
                    Login
                  </button>
                </fieldset>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Auth
